src/main/java/net/openhft/chronicle/queue/impl/single/StoreAppender.java

```diff
@@ -630,21 +645,66 @@ public void writeBytes(final long index, @NotNull final BytesStore bytes) {
+        checkAppendLock();
+        writeLock.lock();
+        try {
+            writeBytesInternal(index, bytes, IndexValidationMode.UNSAFE);
         } finally {
             writeLock.unlock();
         }
     }
 
     /**
-     * Appends bytes without write lock. Should only be used if write lock is acquired externally. Never use without write locking as it WILL corrupt
-     * the queue file and cause data loss.
+     * @deprecated Use {@link #writeBytesInternal(long, BytesStore, IndexValidationMode)} instead
      */
+    @Deprecated(/* To be removed in x.27 */)
     protected void writeBytesInternal(final long index, @NotNull final BytesStore bytes) {
-        writeBytesInternal(index, bytes, false);
+        writeBytesInternal(index, bytes, IndexValidationMode.SAFE);
     }
 
+    /**
+     * @deprecated This method makes no sense when metadata=true, use {@link #writeBytesInternal(long, BytesStore, IndexValidationMode)} instead
+     */
+    @Deprecated(/* To be removed in x.27 */)
     protected void writeBytesInternal(final long index, @NotNull final BytesStore bytes, boolean metadata) {
+        writeBytesInternal(index, bytes, metadata, IndexValidationMode.SAFE);
+    }
+
+    /**
+     * Appends bytes without write lock. Should only be used if write lock is acquired externally. Never use without write locking as it WILL corrupt
+     * the queue file and cause data loss.
+     *
+     * @param index               Index to append at
+     * @param bytes               The excerpt bytes
+     * @param indexValidationMode How stringent to be when validating the append index
+     * @throws IndexOutOfBoundsException when the index specified is not after the end of the queue
+     */
+    protected void writeBytesInternal(final long index, @NotNull BytesStore bytes, IndexValidationMode indexValidationMode) {
+        writeBytesInternal(index, bytes, false, indexValidationMode);
+    }
+
+    /**
+     * Appends bytes without write lock. Should only be used if write lock is acquired externally. Never use without write locking as it WILL corrupt
+     * the queue file and cause data loss.
+     *
+     * @param index               Index to append at
+     * @param bytes               The excerpt bytes
+     * @param metadata            Whether to write the excerpt as metadata, this is legacy and should always be set to false
+     * @param indexValidationMode How stringent to be when validating the append index
+     * @throws IndexOutOfBoundsException when the index specified is not after the end of the queue
+     * @deprecated Use {@link #writeBytesInternal(long, BytesStore, IndexValidationMode)} instead
+     */
+    @SuppressWarnings("DeprecatedIsStillUsed")  // will be collapsed into the caller when deprecated
+    @Deprecated(/* For removal in x.27 */)
+    protected void writeBytesInternal(final long index, @NotNull final BytesStore bytes, boolean metadata, IndexValidationMode indexValidationMode) {
```

Comment:
I've deprecated this method in the interest of removing the metadata parameter.

Firstly I don't think it makes any sense to append a metadata excerpt at a specified index because metadata excerpts don't have indices.

Secondly I believe the check on line 740 would always fail after appending a metadata excerpt, as it will not have incremented the `headerNumber` field (see [here](https://github.com/OpenHFT/Chronicle-Wire/blob/2ca5caf357571eccbdf8cbcfee2ae0438f18e6c1/src/main/java/net/openhft/chronicle/wire/AbstractWire.java#L366)).

So unless anyone objects this change will remove the ability to `writeBytesInternal` a metadata excerpt at a specified index. An ability that arguably doesn't currently exist anyway.

Reply:
Love it

Reply:
Proven. This method never works when `metadata=true`

See https://github.com/OpenHFT/Chronicle-Queue/commit/f4248587d352e9b9d2663ab1665adba6a8fde0f9

Reply:
Fantastic